options/block.go

```diff
@@ -87,22 +63,84 @@ type blockOpts struct{}
 
 var Block blockOpts
 
-// Format is an option for Block.Put which specifies the multicodec to use to
-// serialize the object. Default is "v0"
-func (blockOpts) Format(codec string) BlockPutOption {
+// CidCodec is the modern option for Block.Put which specifies the multicodec to use
+// in the CID returned by the Block.Put operation.
+// It uses correct codes from go-multicodec and replaces the old Format now with CIDv1 as the default.
+func (blockOpts) CidCodec(codecName string) BlockPutOption {
+	return func(settings *BlockPutSettings) error {
+		if codecName == "" {
+			return nil
+		}
+		code, err := codeFromName(codecName)
+		if err != nil {
+			return err
+		}
+		settings.CidPrefix.Codec = uint64(code)
+		return nil
+	}
+}
+
+// Map string to code from go-multicodec
+func codeFromName(codecName string) (mc.Code, error) {
+	var cidCodec mc.Code
+	err := cidCodec.Set(codecName)
+	return cidCodec, err
+}
+
+// Format is a legacy option for Block.Put which specifies the multicodec to
+// use to serialize the object.
+// Provided for backward-compatibility only. Use CidCodec instead.
+func (blockOpts) Format(format string) BlockPutOption {
 	return func(settings *BlockPutSettings) error {
-		settings.Codec = codec
+		if format == "" {
+			return nil
+		}
+		// Opt-in CIDv0 support for backward-compatibility
+		if format == "v0" {
+			settings.CidPrefix.Version = 0
+		}
+
+		// Fixup a legacy (invalid) names for dag-pb (0x70)
+		if format == "v0" || format == "protobuf" {
+			format = "dag-pb"
+		}
+
+		// Fixup invalid name for dag-cbor (0x71)
+		if format == "cbor" {
+			format = "dag-cbor"
+		}
```

Comment:
:information_source: this maintains backward-compatibility of `ipfs block put --format`  in https://github.com/ipfs/go-ipfs/pull/8568